python/social_bot/teacher.py

```diff
@@ -145,6 +148,32 @@ def add_task_group(self, task_group, weight=1):
         self._task_groups.append(task_group)
         self._weights.append(weight)
 
+    def build_vocab_from_tasks(self):
+        vocab_list = []
+        for g in self._task_groups:
+            for t in g._tasks:
+                vocab_list = vocab_list + t.task_vocab
+        # remove repeated words and convert to dict
+        self._vocab_list = sorted(set(vocab_list),key=vocab_list.index)
+        self.vocab_size = len(self._vocab_list)
```

Comment:
need to +1 to include the "0" for padding.

Reply:
operation '+1' is done when building the dict
`        self._vocab_dict = dict(
            zip(self._vocab_list, list(range(1, 1 + self.vocab_size))))`

Reply:
Yes, we should.